pkg/pgmodel/sql_test.go

```diff
@@ -446,136 +519,191 @@ func TestPGXInserterInsertSeries(t *testing.T) {
+					{samples: make([]prompb.Sample, 1)},
+					{samples: make([]prompb.Sample, 1)},
+					{samples: make([]prompb.Sample, 1)},
+					{samples: make([]prompb.Sample, 1)},
+					{samples: make([]prompb.Sample, 1)},
+				},
+			},
+			sqlQueries: []sqlQuery{
+				{sql: "CALL _prom_catalog.finalize_metric_creation()"},
+				{
+					sql:     "SELECT table_name, possibly_new FROM _prom_catalog.get_or_create_metric_table_name($1)",
+					args:    []interface{}{"metric_0"},
+					results: rowResults{},
+					err:     fmt.Errorf("create table error"),
+				},
+			},
 		},
 		{
-			name:           "Not all data inserted",
-			rows:           createRows(5),
-			copyFromResult: 4,
+			name: "Copy from error",
+			rows: map[string][]samplesInfo{
+				"metric_0": {
+					{samples: make([]prompb.Sample, 1)},
+					{samples: make([]prompb.Sample, 1)},
+					{samples: make([]prompb.Sample, 1)},
+					{samples: make([]prompb.Sample, 1)},
+					{samples: make([]prompb.Sample, 1)},
+				},
+			},
+
+			sqlQueries: []sqlQuery{
+				{sql: "CALL _prom_catalog.finalize_metric_creation()"},
+				{
+					sql:     "SELECT table_name, possibly_new FROM _prom_catalog.get_or_create_metric_table_name($1)",
+					args:    []interface{}{"metric_0"},
+					results: rowResults{{"metric_0", true}},
+					err:     error(nil),
+				},
+				{
+					sql: `INSERT INTO "prom_data"."metric_0"(time, value, series_id) SELECT * FROM unnest($1::TIMESTAMPTZ[], $2::DOUBLE PRECISION[], $3::BIGINT[]) a(t,v,s) ORDER BY s,t ON CONFLICT DO NOTHING`,
+					args: []interface{}{
+						[]time.Time{time.Unix(0, 0), time.Unix(0, 0), time.Unix(0, 0), time.Unix(0, 0), time.Unix(0, 0)},
+						make([]float64, 5),
+						make([]int64, 5),
+					},
+					results: rowResults{},
+					err:     fmt.Errorf("some INSERT error"),
+				},
+			},
 		},
 		{
-			name:        "Can't find/create table in DB",
-			rows:        createRows(5),
-			queryNoRows: true,
+			name: "Can't find/create table in DB",
+			rows: map[string][]samplesInfo{
+				"metric_0": {
+					{samples: make([]prompb.Sample, 1)},
+					{samples: make([]prompb.Sample, 1)},
+					{samples: make([]prompb.Sample, 1)},
+					{samples: make([]prompb.Sample, 1)},
+					{samples: make([]prompb.Sample, 1)},
+				},
+			},
+			sqlQueries: []sqlQuery{
+				{sql: "CALL _prom_catalog.finalize_metric_creation()"},
+				{
+					sql:  "SELECT table_name, possibly_new FROM _prom_catalog.get_or_create_metric_table_name($1)",
+					args: []interface{}{"metric_0"},
+					// no results is deliberate
+					results: rowResults{},
+					err:     error(nil),
+				},
+			},
 		},
 		{
-			name:          "Metrics get error",
-			rows:          createRows(1),
+			name: "Metrics get error",
+			rows: map[string][]samplesInfo{
+				"metric_0": {{samples: make([]prompb.Sample, 1)}},
+			},
 			metricsGetErr: fmt.Errorf("some metrics error"),
-		},
-		{
-			name:          "Metrics set error",
-			rows:          createRows(1),
-			metricsSetErr: fmt.Errorf("some metrics error"),
```

Comment:
Failing to set the metric table name is now [explicitly not an error](https://github.com/timescale/promscale/blob/1c852ea723e19b1a4bd63f3c8b00b3fe905c6a09/pkg/pgmodel/sql_ingest.go#L421-L422l) (frankly, I think the metric cache should no longer be returning errors anyway, I just never go around to removing that, partly due to these tests.)